### PR TITLE
Scheduling a call over sms

### PR DESCRIPTION
**Pull request by oliverbarnes**
Ready for review

### CODE REVIEW

File: Guardfile [hunk 1]
```diff
@@ -38,31 +38,48 @@ group :red_green_refactor, halt_on_fail: true do
...
-    watch('app/mailers/event_invitation_mailer.rb') { 'spec/features/person_responds_to_interview_invitation_spec.rb' }
-    watch('app/models/v2/reservation.rb') { 'spec/features/person_responds_to_interview_invitation_spec.rb' }
-    watch('app/controllers/v2/reservations_controller.rb') { 'spec/features/person_responds_to_interview_invitation_spec.rb' }
-    watch('app/views/v2/reservations/new.html.erb') { 'spec/features/person_responds_to_interview_invitation_spec.rb' }
-    watch('app/views/v2/reservations/_form.html.erb') { 'spec/features/person_responds_to_interview_invitation_spec.rb' }
+    watch('app/views/event_invitation_mailer/invite.text.erb') { 'spec/features/person_responds_to_interview_invitation_over_email_spec.rb' }
+    watch('app/views/event_invitation_mailer/invite.text.erb') { 'spec/features/sms_invitation_to_phone_call_spec.rb' }
+    watch('app/views/public/people/new.html.erb') { 'spec/features/people_registration_spec.rb' }
+    watch('app/views/public/people/_form.html.erb') { 'spec/features/people_registration_spec.rb' }
   end
```
Comment (marcoow) [inline Guardfile]: Doesn't seem to be a sustainable approach to list so many individual files with their individual dependencies here. I'd rather only list patterns and run a few tests that might not be necessary to run each time than having to keep this list up to date.

File: app/controllers/v2/sms_reservations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    end
+
+    def send_error_notification
+      ::InvalidOptionSms.new(to: reservation_params[:From]).send
+
+      render text: 'OK'
+    end
+
+    def only_numbers_and_at_least_two_of_them?
+      message =~ /^\d\d+$/
```
Comment (marcoow) [inline app/controllers/v2/sms_reservations_controller.rb]: I guess this should allow spaces as well

File: app/sms/application_sms.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class ApplicationSms
+  attr_reader :client, :application_number
+
+  def initialize(*)
```
Comment (marcoow) [inline app/sms/application_sms.rb]: what's the `*`?

File: Guardfile (hunk 1 above)
Comment (oliverbarnes) [inline Guardfile]: Yeah, I agree it's pretty unwieldy. That's something I've been debating each time I update this, as well. A lot of the individual entries are for feature specs which have non-conventional naming, and so are hard to use patterns with. Given the time constraints, feature specs are the only coverage for certain files, and so the only things that run on guard when updating them. So without the individual entries, with these files it's like not using guard at all. I end up running the feature specs by hand after updating them.

I also really like getting immediate feedback from all the features affected when changing a  dependency.

Maybe we can find some middle ground on how to do this, conventions for naming or file org we could use to match patterns against?

@cromulus any thoughts on this?

File: app/controllers/v2/sms_reservations_controller.rb (hunk 2 above)
Comment (oliverbarnes) [inline app/controllers/v2/sms_reservations_controller.rb]: adding this

File: app/sms/application_sms.rb (hunk 3 above)
Comment (oliverbarnes) [inline app/sms/application_sms.rb]: alternative to `*args` when one wants to allow any args and not use them specifically (in this case, only subclasses accept args). rubocop likes it better

**Comment by cromulus** [comment]
FYI, communication preferences are an enum: 
"2" == "Email"
"1" == "SMS"

Not sure if that was captured, but was news to me getting things setup.

**Pull request closed without merging**